### PR TITLE
Adds permissions for using workflows and content scheduling

### PR DESCRIPTION
**Pull request by stephen-cox**
Part of:
* https://github.com/localgovdrupal/localgov_core/issues/92
* https://github.com/localgovdrupal/localgov_core/issues/96

### CODE REVIEW

File: modules/localgov_services_status/localgov_services_status.module [hunk 1]
```diff
@@ -37,6 +38,63 @@ function localgov_services_status_theme($existing, $type, $theme, $path) {
...
+    ],
+    RolesHelper::CONTRIBUTOR_ROLE => [
+      'create localgov_services_status content',
+      'delete own localgov_services_status content',
+      'edit any localgov_services_status content',
+      'edit own localgov_services_status content',
+      'revert localgov_services_status revisions',
+      'view localgov_services_status revisions',
+    ],
+  ];
```
Comment (andybroomfield) [inline modules/localgov_services_status/localgov_services_status.module]: Should this block not also be wrapped in `if (\Drupal::moduleHandler()->moduleExists('localgov_roles')) {` ?

Is including localgov_roles as an explicit dependacy now aggreed?  BHCC already has content roles and would rather not have to go and then delete them just becuase we use the service modules. Also would these break if BHCC did not have the defined content editor roles.

Comment (stephen-cox) [reply modules/localgov_services_status/localgov_services_status.module]: This code is called by the hook_localgov_roles_default() hook, which is part of localgov_roles. So these permissions only get applied if localgov_roles is enabled.

Comment (andybroomfield) [reply modules/localgov_services_status/localgov_services_status.module]: Ok, that makes sense thanks.
So to confirm, if BHCC does not use localgov_roles (yet) that is ok?

Comment (ekes) [reply modules/localgov_services_status/localgov_services_status.module]: The `use RolesHelper` statement is still parsed though.

Even if it's in the default code base I never tested if Drupal's autoloader compiler includes disabled modules, so used the fully pathed namesapce (and added a coding standards ignore). If there's a chance it's not in the codebase as it's not used then even more so.

File: modules/localgov_services_status/localgov_services_status.module [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 use Drupal\Core\Cache\Cache;
 use Drupal\Core\Url;
+use Drupal\localgov_roles\RolesHelper;
```
Comment (andybroomfield) [inline modules/localgov_services_status/localgov_services_status.module:10]: Will this use statement work if localgov_roles is not installed and enabled?
(Assume the code will exist becuase localgov_core is a dependancy).

Comment (stephen-cox) [reply modules/localgov_services_status/localgov_services_status.module:10]: Works fine as long as localgov_core is installed. All modules that define content types have a dependency on core so this use statement works fine for the PRs I'm in the process of creating.

File: modules/localgov_services_status/localgov_services_status.module (hunk 1 above)
Comment (stephen-cox) [reply modules/localgov_services_status/localgov_services_status.module]: > So to confirm, if BHCC does not use localgov_roles (yet) that is ok?

Yep - enabling localgov_roles and then enabling these modules or localgov_workflows will bring in all the permissions, but if you don't have roles enabled you won't get the permissions set

**Review by finnlewis — APPROVED** [review]
I'm happy with this. 

Tested locally, tried installing in different orders and get consistent results, the permissions are added for the roles if they exist. 

I've been testing the same thing on each project too, for example I've reviewed and approved : 

https://github.com/localgovdrupal/localgov_guides/pull/65
https://github.com/localgovdrupal/localgov_step_by_step/pull/42
https://github.com/localgovdrupal/localgov_subsites/pull/51

**Pull request merged**